### ISSUE TITLE
Improve check-in time

### ISSUE THREAD

**Issue by BalestraPatrick** [swissmobidevs/appbuilders16#4]
It took us 1 hour to check-in about 200 people. We had planned to check-in everyone in 30 minutes and that meant that each person should have taken 9 seconds. 
Finding the correct badge for each speaker was the operation that slowed us down. We should split the badges by last name and order them alphabetically. Having more staff members during this crucial phase would also improve the duration of the check-in.

**Comment by fpillet** [comment]
Also maybe start check-in a bit earlier? Unclogs the line for people who arrive early, so you don't have to put more staff

**Comment by D-32** [comment]
I agree with @BalestraPatrick, the process of finding the right card took way too long. 

Tshirts could have been giving away at a separate table, for instance just after you walk in. We had enough people, but being 3-4 people at the registration desk all doing different tasks didn't help. If somebody would just do the tshirts at a different location, some mess would have already been reduced.

We did two tables and split up A-K & L-Z. Next time though we should sign post that at the beginning of the queue so that people already split up and form two queues.

**Comment by akosma** [comment]
I'd suggest the following points, each with its mandatory geek reference:
1. **Fail fast:** Open the check-in earlier, 8 AM for example – or even maybe the day before if the venue allows it, à la WWDC.
2. **Avoid polling:** Self check-in iOS app on an iPad Pro that uses the front camera to read the QR code and displays the pic of the person and a sound when the checkin is done. Use several of those if possible. Small REST service behind to gather the info.
3. **GCD with queues:** For 200 attendees, separate badges in three categories: family names starting with A-H, I-P and those starting with Q-Z. Have at most 60 badges per category, if you have more attendees.
4. **Add an index to the database, and avoid table scans:** Place the badges on a table, sorted in alphabetical order, using the family name as a basis; people will find it themselves and pick it up.
5. **Defer:** Giving the t-shirt and goodie bag can happen at any time, it does not have to be at the same time as the registration itself.
6. **Subclass:** Separate desk for speakers and staff with a more manual approach, if required.

**Comment by akosma** [comment]
You know, "premature optimisation is the root of all evil" and all that. Boom.
